### DISCUSSION TITLE
How to disallow access to AdminUI for normal users

### DISCUSSION THREAD

**Discussion by theskillwithin** [keystonejs/keystone#7775] [Questions] [answered]
I know after setting permissions it doesn't matter if a user can login to AdminUI or not in theory.  but it would be nice to be able to disallow people from logging into adminui while still having access the the api's from

**Comment by flexdinesh** [comment, marked as answer]
@theskillwithin Do you mean disallow only certain types of users from logging into Admin UI or disable Admin UI for everyone?

If you want to disallow certain types of users, you can add a field called `isAdmin` or anything relevant to User list and prevent users who are not admins from logging into the Admin UI using access control.

If you want to disable Admin UI for everyone, it's available as a [ui config](https://keystonejs.com/docs/apis/config#ui).
```
ui: {
  isDisabled: true
}
```

**Reply by theskillwithin** [reply]
Talking about only disallowing certain types of users from logging into AdminUI

**Reply by flexdinesh** [reply]
You will need to setup auth and access control - https://keystonejs.com/docs/guides/auth-and-access-control first. 

Then you can configure how to prevent non-admins from logging in using a single line —
```
ui: {
  // Allow Admin UI access only for admins
  isAccessAllowed: ({ session }) => !!session?.data?.isAdmin, // or anything equivalent
},
```
https://keystonejs.com/docs/apis/config#ui

**Reply by flexdinesh** [reply]
Here's a codesandbox example with auth + access control to only allow admins to login to Admin UI — https://codesandbox.io/s/auth-access-control-uk06ez?file=/keystone.ts
